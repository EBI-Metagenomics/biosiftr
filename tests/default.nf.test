nextflow_pipeline {

    name "Test bioSIFTR pipeline main.nf"
    script "../main.nf"
    config "./nextflow.config"
    tag "pipeline"


    test("bioSIFTR running on single-end and paired-end reads. Biome human-vaginal-v1-0 default params") {
        when {
            params {
                input = "${baseDir}/tests/test_samplesheet.csv"
                biome = "human-vaginal-v1-0"
                outdir = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(
                params.outdir, 
                relative: true, 
                includeDir: true, 
                ignore: [
                    'pipeline_info/*'
                ]
            )
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name
                    stable_name,
                    // All files with stable contents
                    stable_path,
                ).match() }
            )
        }
    }

    test("bioSIFTR running on single-end and paired-end reads. Biome human-vaginal-v1-0 with bwa, core, and dram options") {
        when {
            params {
                input = "${baseDir}/tests/test_samplesheet.csv"
                biome = "human-vaginal-v1-0"
                outdir = "$outputDir"
                run_bwa = true
                core_mode = true
                run_dram = true
            }
        }
        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(
                params.outdir, 
                relative: true, 
                includeDir: true, 
                ignore: [
                    'pipeline_info/*'
                ]
            )
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name
                    stable_name,
                    // All files with stable contents
                    stable_path,
                ).match() }
            )
        }
    }
}
