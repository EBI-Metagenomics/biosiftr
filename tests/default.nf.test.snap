{
    "bioSIFTR running on single-end and paired-end reads. Biome human-vaginal-v1-0 with bwa, core, and dram options": {
        "content": [
            28,
            [
                "dram_results",
                "dram_results/integrated_bwa_community_0_dram.html",
                "dram_results/integrated_bwa_community_dram.tsv",
                "dram_results/integrated_sm_community_0_dram.html",
                "dram_results/integrated_sm_community_dram.tsv",
                "dram_results/test_bwa_species_0_dram.html",
                "dram_results/test_bwa_species_dram.tsv",
                "dram_results/test_bwa_species_dram_summary.tsv",
                "dram_results/test_sm_species_0_dram.html",
                "dram_results/test_sm_species_dram.tsv",
                "dram_results/test_sm_species_dram_summary.tsv",
                "dram_results/versions.yml",
                "function_tables",
                "function_tables/test_bwa_community_kegg.tsv",
                "function_tables/test_bwa_community_pfams.tsv",
                "function_tables/test_bwa_species_kegg.tsv",
                "function_tables/test_bwa_species_pfams.tsv",
                "function_tables/test_sm_community_kegg.tsv",
                "function_tables/test_sm_community_pfams.tsv",
                "function_tables/test_sm_species_kegg.tsv",
                "function_tables/test_sm_species_pfams.tsv",
                "integrated_annotation",
                "integrated_annotation/bwa_kos_matrix.tsv",
                "integrated_annotation/bwa_modules_matrix.tsv",
                "integrated_annotation/bwa_pfam_matrix.tsv",
                "integrated_annotation/bwa_taxo_matrix.tsv",
                "integrated_annotation/sm_kos_matrix.tsv",
                "integrated_annotation/sm_modules_matrix.tsv",
                "integrated_annotation/sm_pfam_matrix.tsv",
                "integrated_annotation/sm_taxo_matrix.tsv",
                "kegg_completeness",
                "kegg_completeness/test_bwa_community_kegg_modules_comp.tsv",
                "kegg_completeness/test_bwa_species_kegg_modules_comp.tsv",
                "kegg_completeness/test_sm_community_kegg_modules_comp.tsv",
                "kegg_completeness/test_sm_species_kegg_modules_comp.tsv",
                "mapping",
                "mapping/bwamem2",
                "mapping/bwamem2/test_0.1_u_relab.tsv",
                "mapping/sourmash",
                "mapping/sourmash/test.csv.gz",
                "pipeline_info",
                "quality_control",
                "quality_control/decont_fastqc",
                "quality_control/decont_fastqc/test_1_fastqc.html",
                "quality_control/decont_fastqc/test_2_fastqc.html",
                "quality_control/fastp",
                "quality_control/fastp/test.fastp.html",
                "quality_control/fastp/test.fastp.json",
                "quality_control/fastp/test.fastp.log",
                "quality_control/fastp/test_1.fastp.fastq.gz",
                "quality_control/fastp/test_2.fastp.fastq.gz",
                "quality_control/fastp/versions.yml",
                "quality_control/multiqc",
                "quality_control/multiqc/multiqc_data",
                "quality_control/multiqc/multiqc_data/mqc_fastp-insert-size-plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_1_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_2_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp_filtered_reads_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_base_n_content_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_base_sequence_quality_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_sequence_gc_content_plot_Counts.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_sequence_quality_scores_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_sequence_counts_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_sequence_duplication_levels_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_sequence_length_distribution_plot_1.txt",
                "quality_control/multiqc/multiqc_data/multiqc.log",
                "quality_control/multiqc/multiqc_data/multiqc_citations.txt",
                "quality_control/multiqc/multiqc_data/multiqc_data.json",
                "quality_control/multiqc/multiqc_data/multiqc_fastp.txt",
                "quality_control/multiqc/multiqc_data/multiqc_fastqc.txt",
                "quality_control/multiqc/multiqc_data/multiqc_general_stats.txt",
                "quality_control/multiqc/multiqc_data/multiqc_sources.txt",
                "quality_control/multiqc/multiqc_plots",
                "quality_control/multiqc/multiqc_plots/pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-insert-size-plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_1_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_2_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp_filtered_reads_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp_filtered_reads_plot_1_pc.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_base_n_content_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_base_sequence_quality_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_quality_scores_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_counts_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_counts_plot_1_pc.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_duplication_levels_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_length_distribution_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-insert-size-plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_1_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_2_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp_filtered_reads_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp_filtered_reads_plot_1_pc.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_base_n_content_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_gc_content_plot_Counts.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_gc_content_plot_Percentages.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_counts_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_counts_plot_1_pc.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_duplication_levels_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_length_distribution_plot_1.png",
                "quality_control/multiqc/multiqc_plots/svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-insert-size-plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_1_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_2_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp_filtered_reads_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp_filtered_reads_plot_1_pc.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_base_n_content_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_base_sequence_quality_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_gc_content_plot_Counts.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_quality_scores_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_counts_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_counts_plot_1_pc.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_duplication_levels_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_length_distribution_plot_1.svg",
                "quality_control/multiqc/multiqc_report.html",
                "taxonomy_tables",
                "taxonomy_tables/test_bwa_species.tsv",
                "taxonomy_tables/test_sm_species.tsv"
            ],
            [
                "test.csv.gz:md5,c9f515d9c808531b9b8cb6756359e466",
                "test_bwa_species.tsv:md5,0e4c23d7ec39409a36c7e685222c07a3",
                "test_sm_species.tsv:md5,179b7dd3bad95009468cf50e066097d4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-24T10:44:17.459533927"
    },
    "bioSIFTR running on single-end and paired-end reads. Biome human-vaginal-v1-0 default params": {
        "content": [
            15,
            [
                "dram_results",
                "function_tables",
                "function_tables/test_sm_community_kegg.tsv",
                "function_tables/test_sm_community_pfams.tsv",
                "function_tables/test_sm_species_kegg.tsv",
                "function_tables/test_sm_species_pfams.tsv",
                "integrated_annotation",
                "integrated_annotation/sm_kos_matrix.tsv",
                "integrated_annotation/sm_modules_matrix.tsv",
                "integrated_annotation/sm_pfam_matrix.tsv",
                "integrated_annotation/sm_taxo_matrix.tsv",
                "kegg_completeness",
                "kegg_completeness/test_sm_community_kegg_modules_comp.tsv",
                "kegg_completeness/test_sm_species_kegg_modules_comp.tsv",
                "mapping",
                "mapping/sourmash",
                "mapping/sourmash/test.csv.gz",
                "pipeline_info",
                "quality_control",
                "quality_control/decont_fastqc",
                "quality_control/decont_fastqc/test_1_fastqc.html",
                "quality_control/decont_fastqc/test_2_fastqc.html",
                "quality_control/fastp",
                "quality_control/fastp/test.fastp.html",
                "quality_control/fastp/test.fastp.json",
                "quality_control/fastp/test.fastp.log",
                "quality_control/fastp/test_1.fastp.fastq.gz",
                "quality_control/fastp/test_2.fastp.fastq.gz",
                "quality_control/fastp/versions.yml",
                "quality_control/multiqc",
                "quality_control/multiqc/multiqc_data",
                "quality_control/multiqc/multiqc_data/mqc_fastp-insert-size-plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_1_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_2_After_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastp_filtered_reads_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_base_n_content_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_base_sequence_quality_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_sequence_gc_content_plot_Counts.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_per_sequence_quality_scores_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_sequence_counts_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_sequence_duplication_levels_plot_1.txt",
                "quality_control/multiqc/multiqc_data/mqc_fastqc_sequence_length_distribution_plot_1.txt",
                "quality_control/multiqc/multiqc_data/multiqc.log",
                "quality_control/multiqc/multiqc_data/multiqc_citations.txt",
                "quality_control/multiqc/multiqc_data/multiqc_data.json",
                "quality_control/multiqc/multiqc_data/multiqc_fastp.txt",
                "quality_control/multiqc/multiqc_data/multiqc_fastqc.txt",
                "quality_control/multiqc/multiqc_data/multiqc_general_stats.txt",
                "quality_control/multiqc/multiqc_data/multiqc_sources.txt",
                "quality_control/multiqc/multiqc_plots",
                "quality_control/multiqc/multiqc_plots/pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-insert-size-plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_1_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_2_After_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp_filtered_reads_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastp_filtered_reads_plot_1_pc.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_base_n_content_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_base_sequence_quality_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_per_sequence_quality_scores_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_counts_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_counts_plot_1_pc.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_duplication_levels_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/pdf/mqc_fastqc_sequence_length_distribution_plot_1.pdf",
                "quality_control/multiqc/multiqc_plots/png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-insert-size-plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_1_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_2_After_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp_filtered_reads_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastp_filtered_reads_plot_1_pc.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_base_n_content_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_base_sequence_quality_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_gc_content_plot_Counts.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_gc_content_plot_Percentages.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_per_sequence_quality_scores_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_counts_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_counts_plot_1_pc.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_duplication_levels_plot_1.png",
                "quality_control/multiqc/multiqc_plots/png/mqc_fastqc_sequence_length_distribution_plot_1.png",
                "quality_control/multiqc/multiqc_plots/svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-insert-size-plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_1_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_1_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_2_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-gc-plot_Read_2_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_1_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_1_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_2_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-content-n-plot_Read_2_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_1_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_1_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_2_After_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp-seq-quality-plot_Read_2_Before_filtering.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp_filtered_reads_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastp_filtered_reads_plot_1_pc.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_base_n_content_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_base_sequence_quality_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_gc_content_plot_Counts.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_per_sequence_quality_scores_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_counts_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_counts_plot_1_pc.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_duplication_levels_plot_1.svg",
                "quality_control/multiqc/multiqc_plots/svg/mqc_fastqc_sequence_length_distribution_plot_1.svg",
                "quality_control/multiqc/multiqc_report.html",
                "taxonomy_tables",
                "taxonomy_tables/test_sm_species.tsv"
            ],
            [
                "test.csv.gz:md5,c9f515d9c808531b9b8cb6756359e466",
                "test_sm_species.tsv:md5,179b7dd3bad95009468cf50e066097d4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-06-23T15:26:09.573568348"
    }
}